Library/Homebrew/cmd/install.rb

```diff
@@ -14,8 +14,8 @@ def install
     end
 
     ARGV.named.each do |name|
-      if !File.exist?(name) && (name =~ HOMEBREW_TAP_FORMULA_REGEX \
-                                || name =~ HOMEBREW_CASK_TAP_FORMULA_REGEX)
+      if !File.exist?(name) && (!name =~ HOMEBREW_CORE_FORMULA_REGEX) \
```

Comment:
You don't need to test  `HOMEBREW_CORE_FORMULA_REGEX` here as you cannot/should not install core tap.